README.md

```diff
@@ -59,8 +59,16 @@ Once you have implemented a lower bound search for the start of the range, imple
 *Answer the questions below and then push this file to GitHub.*
 
 *How do you use binary search to find the lower bound of a number? How did you have to modify the binary search algorithm?*
+In binary search, we normally get some value of x , and we stop the search when we find it or return False if the value x is not present in the list.
+In our implementation, we loop until the low and high bounds are the same. 
+Like that, we can return the start where the integer that I am looking for is located.
 
 *Would anything be more difficult if the features covered ranges instead of single nucleotides (like real BED files)? What could go wrong, if anything?*
+Yes, our implementation would not work correctly. 
+We could have an overlapping region, and we would need to add some new conditions about the start and end of chromosomes.
 
 *We wrote a tool for merging two BED files, but what if we had a bunch of them? What would the complexity be if we merged them in, one at a time? What would the complexity be if we merged all of the files at the same time?*
-
+The complexity of the merge we have right now is O(N+M), where N and M are the lenghts of the two files.
+If we merged the bed files one by one, it would just be O(N+M+O+..) where N and M and P are the lenghts of our files.
+If we merged the bed files all at once it should take the same amount of time, as there is the same number of comparisons,
+but it would probably be harder to implement.
```

Comment:
Let’s say the first two files had sizes $n_1$ and $n_2$, respectively. It would take $O(n_1+n_2)$ to merge them. Then, the third file, with size $n_3$ comes along and needs to be merged into that. Now we have one file of size $n_1+n_2$ and one of size $n_3$ so that would take time $O((n_1+n_2)+n_3)$. So far, that looks like your calculation, but this is only the time it takes to merge the _third_ file in, not merging all three of them; that would take $O(n_1+n2 + (n_1+n_2)+n_3)=O(2(n_1+n_2)+n_3)$. Here, the $O(-)$-notation rules seem to break down, because the rule of thumb is that $O(k\cdot f(n))=O(f(n))$ for any constant $k$, but that is because $k$ _isn’t_ a constant when we continue. When we add the fourth file, with size $n_4$, we need to merge it with a file of size $n_1+n_2+n_3$, which takes $O(n_1+n_2+n_3+n_4)$ but that is in addition to the $O(2(n_1+n_2)+n_3)$ we already spent for merging the first three, so the actual time to merge four would be $O(3(n_1+n_2)+2n_3+n_4)$.

You can continue this expansion and see that when we have merged $m$ files, we have run through $n_1$ and $n_2$ $m-1$ times, $n_3$ $m-2$ times and so on. This running time is _a lot_ worse than if you could merge all the files at once.

If you don’t want to merge all files at once (and it isn’t at all easy to or free to find the smallest in a list of length $m$ either), you can merge them pairwise. That means that each single file’s elements will only be used in $O(\log m)$ different merge operations, which is a lot more efficient.